src/H5VL_log_dataset.cpp

```diff
@@ -65,7 +65,7 @@ void *H5VL_log_dataset_create (void *obj,
     int i;
     H5VL_log_obj_t *op        = (H5VL_log_obj_t *)obj;
     H5VL_log_dset_t *dp       = NULL;
-    H5VL_log_dset_info_t *dip = NULL;  // Dataset info
+    H5VL_log_dset_info_t *dip = nullptr;  // Dataset info
```

Comment:
I wonder if it would be worth separating the C++ from the C to avoid directly mixing `NULL` and `nullptr` near each other?

Reply:
Updated to the use of nullptr  within the same function.